digits/model/tasks/caffe_train.py

```diff
@@ -1472,3 +1415,178 @@ def net_sanity_check(net, phase):
                                                      "this blob is not included at that stage. Please consider " \
                                                      "using an include directive to limit the scope of this layer." % (
                                                        layer.name, bottom, "TRAIN" if phase == caffe_pb2.TRAIN else "TEST"))
+
+
+def cleanedUpClassificationNetwork(original_network, num_categories):
+    """
+    Perform a few cleanup routines on a classification network
+    Returns a new NetParameter
+    """
+    network = caffe_pb2.NetParameter()
+    network.CopyFrom(original_network)
+
+    for layer in network.layer:
+        if 'Data' in layer.type:
+            assert layer.type in ['Data', 'HDF5Data'], \
+                'Unsupported data layer type %s' % layer.type
+
+        if layer.type == 'Accuracy':
+            # Check to see if top_k > num_categories
+            if ( layer.accuracy_param.HasField('top_k') and
+                    layer.accuracy_param.top_k >= num_categories ):
+                self.logger.warning(
```

Comment:
Oops I misspoke. The way I implemented it will break if two subsequent layers both have an invalid top_k because the second one won't be processed.

I need to fix that tomorrow...

```python
>>> a = range(10)
>>> for i, x in enumerate(a):
...   print 'Processing %d (%d) ...' % (i, x)
...   if (x%3 == 0):
...     del a[i]
...
Processing 0 (0) ...
Processing 1 (2) ...
Processing 2 (3) ...
Processing 3 (5) ...
Processing 4 (6) ...
Processing 5 (8) ...
Processing 6 (9) ...
>>> a
[1, 2, 4, 5, 7, 8]
```